go-controller/pkg/ovn/kubevirt.go

```diff
@@ -0,0 +1,88 @@
+package ovn
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	kapi "k8s.io/api/core/v1"
+
+	kvv1 "kubevirt.io/api/core/v1"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kubevirt"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	ovntypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+func (oc *DefaultNetworkController) ensureDHCPOptionsForVM(pod *corev1.Pod, lsp *nbdb.LogicalSwitchPort) error {
+	if !kubevirt.PodIsLiveMigratable(pod) {
+		return nil
+	}
+	ovnPodAnnotation, err := util.UnmarshalPodAnnotation(pod.Annotations, ovntypes.DefaultNetworkName)
+	if err != nil {
+		return fmt.Errorf("failed retrieving subnets to configure DHCP at lsp %s: %v", lsp.Name, err)
+	}
+	// Fake router to delegate on proxy arp mechanism
+	vmName, ok := pod.Labels[kvv1.VirtualMachineNameLabel]
+	if !ok {
+		return fmt.Errorf("missing %s label at pod %s/%s when configuaring DHCP", kvv1.VirtualMachineNameLabel, pod.Namespace, pod.Name)
+	}
+	dhcpConfig, err := kubevirt.ComposeDHCPConfig(oc.watchFactory, vmName, ovnPodAnnotation.IPs)
+	if err != nil {
+		return fmt.Errorf("failed composing DHCP options: %v", err)
+	}
+	var (
+		dhcpv4OptionsPredicate, dhcpv6OptionsPredicate func(*nbdb.DHCPOptions) bool
+	)
+	if dhcpConfig.V4Options != nil {
+		dhcpv4OptionsDbObjectID := libovsdbops.NewDbObjectIDs(libovsdbops.VirtualMachineDHCPOptions, oc.controllerName,
+			map[libovsdbops.ExternalIDKey]string{
+				libovsdbops.ObjectNameKey: dhcpConfig.V4Options.Cidr,
+				libovsdbops.HostnameIndex: vmName,
+			})
+		dhcpConfig.V4Options.ExternalIDs = dhcpv4OptionsDbObjectID.GetExternalIDs()
+		dhcpConfig.V4Options.ExternalIDs[kvv1.VirtualMachineNameLabel] = vmName
+		dhcpConfig.V4Options.ExternalIDs[kubevirt.NamespaceExternalIDKey] = pod.Namespace
```

Comment:
could you encapsulate this in an helper func ? I think that would be more readable.